webapp/src/main/resources/public/js/stores/ScreenshotViewerStore.js

```diff
@@ -0,0 +1,48 @@
+class ScreenshotViewerStore {
+
+    constructor() {
+        this.setDefaultState();
+    }
+
+    setDefaultState() {
+        this.show = false;
+        this.src = null;
+        this.number = 0;
+        this.total = 0;
+        this.branchStatisticScreenshots = null;
+        this.textUnits = [];
+    }
+
+    open() {
```

Comment:
Probably make more sense to have `open(branchStatisticScreenshots)` so it makes it clearer what is required for the parent class to work and then you can remove one of the override

Reply:
but one open takes an id and the other takes the list of screenshots, wouldn't this be misleading for the one that takes id?

Reply:
good point the one with Id, should be renamed to `openWithBranchStatisticId ` or something like that